### PR TITLE
[v10.0.x] Docs: Remove old plugin docs, fix URLs to go to new dev portal (#75325)

### PR DESCRIPTION
**Pull request by jdbaldry**
* Remove old plugin docs, add redirects to new dev portal

* Restore plugin.schema.json file

* Update docs/sources/administration/plugin-management/index.md

* More updates

* Cleanup links

* Update docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md

* fix codeowners

* Change reference links to inline links to avoid 404s

* Update docs/sources/datasources/_index.md

* Update docs/sources/datasources/mysql/_index.md

* Update docs/sources/datasources/postgres/_index.md

* Update docs/sources/fundamentals/timeseries-dimensions/index.md

* Update docs/sources/fundamentals/dashboards-overview/index.md

* Update docs/sources/fundamentals/dashboards-overview/index.md

* Update docs/sources/panels-visualizations/configure-standard-options/index.md

* Update docs/sources/panels-visualizations/query-transform-data/transform-data/index.md

* Remove grafana.com from 2 URLs

* Fix 1 more reference link

* Prettier fix

---------

(cherry picked from commit 0e2b741fc33ae5382fb27b4659f2b3ecd13f2afe)

### CODE REVIEW

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 1]
```diff
@@ -974,4 +974,27 @@ Here is the result after adding a Limit transformation with a value of '3':
 This transformation is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file]({{< relref "../../../setup-grafana/configure-grafana/#configuration-file-location" >}}) to enable the `timeSeriesTable` [feature toggle]({{< relref "../../../setup-grafana/configure-grafana/#feature_toggles" >}}) to use it.
 {{% /admonition %}}
 
-Use this transformation to convert time series result into a table, converting time series data frame into a "Trend" field. "Trend" field can then be rendered using [sparkline cell type]({{< relref "../../visualizations/table/#sparkline" >}}), producing an inline sparkline for each table row. If there are multiple time series queries, each will result in a separate table data frame. These can be joined using join or merge transforms to produce a single table with multiple sparklines per row.
+Use this transformation to format the output of a time field. Output can be formatted using (Moment.js format strings)[https://momentjs.com/docs/#/displaying/]. For instance, if you would like to display only the year of a time field the format string `YYYY` can be used to show the calendar year (e.g. 1999, 2012, etc.).
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:977]: Incorrect content
```suggestion
```

**Review by imatwawana — COMMENTED** [review]
Found one problem

**Review by chri2547 — APPROVED** [review]

**Comment by jdbaldry** [comment]
Thanks for spotting that Isabel!

File: docs/sources/administration/data-source-management/index.md [hunk 2]
```diff
@@ -195,4 +195,4 @@ For more documentation on a specific data source plugin's features, including it
 
 ### Create a data source plugin
 
-To build your own data source plugin, refer to the ["Build a data source plugin"](/tutorials/build-a-data-source-plugin/) tutorial and our documentation about [building a plugin](/developers/plugins/).
+To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](/developers/plugins-tools).
```
Comment (josmperez) [inline docs/sources/administration/data-source-management/index.md:198]:
```suggestion
To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](/developers/plugin-tools).
```

File: docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ Grafana managed alerts query the following backend data sources that have alerti
 
 - built-in data sources or those developed and maintained by Grafana: `Graphite`, `Prometheus`, `Loki`, `InfluxDB`, `Elasticsearch`,
   `Google Cloud Monitoring`, `Cloudwatch`, `Azure Monitor`, `MySQL`, `PostgreSQL`, `MSSQL`, `OpenTSDB`, `Oracle`, and `Azure Monitor`
-- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json))
+- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json)
```
Comment (josmperez) [inline docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md:33]:
```suggestion
- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json))
```

**Review by josmperez — APPROVED** [review]
Two nits, overall LGTM.

File: .changelog-archive/CHANGELOG.7.md [hunk 4]
```diff
@@ -1294,7 +1294,7 @@ This option to group query variable values into groups by tags has been an exper
 **Deprecation warnings**
 
 - Scripted dashboards is now deprecated. The feature is not removed but will be in a future release. We hope to address the underlying requirement of dynamic dashboards in a different way. [#24059](https://github.com/grafana/grafana/issues/24059)
-- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](https://grafana.com/docs/grafana/latest/developers/plugins/backend/) for information about the new officially supported backend plugins.
+- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](/developers/plugin-tools) for information about the new officially supported backend plugins.
```
Comment (josmperez) [inline .changelog-archive/CHANGELOG.7.md:1297]:
```suggestion
- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](/developers/plugin-tools/introduction/backend-plugins) for information about the new officially supported backend plugins.
```

**Review by josmperez — APPROVED** [review]
LGTM.

File: docs/sources/developers/_index.md [hunk 5]
```diff
@@ -11,12 +11,18 @@ weight: 190
 
 # Developers
 
-This section includes the following Grafana developer documentation:
+Go to the [Grafana developer portal](/developers) to access the following documentation:
+
+- [Grafana plugin development](/developers/plugin-tools)
+- [Grafana design system](https://developers.grafana.com)
```
Comment (josmperez) [inline docs/sources/developers/_index.md:17]:
```suggestion
- [Grafana UI component library](https://developers.grafana.com)
```

**Pull request merged**